### PR TITLE
[SDK][PSEH] Improve PSEH a bit

### PR DESCRIPTION
**Pull request by tkreuzer**
Add PSEH include dir globally, include pseh2.h from excpt.h and add compatibility macros to reduce hacks in 3rd-perty code.

## Tests
✅ KVM x86: https://reactos.org/testman/compare.php?ids=98835,98839,98874,98877
✅ KVM x64: https://reactos.org/testman/compare.php?ids=98846,98850,98883,98886

### CODE REVIEW

File: sdk/lib/pseh/include/pseh/pseh2_64.h [hunk 1]
```diff
@@ -173,3 +173,12 @@ __seh2$$begin_except__: __MINGW_ATTRIB_UNUSED;
 #define _SEH2_LEAVE goto __seh2$$leave_scope__
 #define _SEH2_YIELD(__stmt) __stmt
 #define _SEH2_VOLATILE volatile
+
+#define __try _SEH2_TRY
```
Comment (HBelusca) [inline sdk/lib/pseh/include/pseh/pseh2_64.h:177]: No `#ifndef __try` needed in the x64 version ?

**Review by HBelusca — APPROVED** [review]

File: sdk/lib/pseh/include/pseh/pseh2_64.h (hunk 1 above)
Comment (tkreuzer) [reply sdk/lib/pseh/include/pseh/pseh2_64.h:178]: In fact, yes. It just passed, because we allow warnings on x64 builds. Fixed it.

File: CMakeLists.txt [hunk 2]
```diff
@@ -324,7 +324,8 @@ Enable this if the module uses typeid or dynamic_cast. You will probably need to
         sdk/include/ddk
         sdk/include/ndk
         sdk/include/reactos
-        sdk/include/reactos/libs)
+        sdk/include/reactos/libs
+        sdk/lib/pseh/include)
```
Comment (learn-more) [inline CMakeLists.txt:328]: Why did you add it globally instead of `public` on the `pseh` library?
Then only projects linking to `pseh` will get the include

Comment (tkreuzer) [reply CMakeLists.txt:328]: That has already been a problem in the past, see my previous fix to usetup: To use PSEH in setuplib, setuplib needs to link to pseh library, which in turn will cause usetup to link to link to pseh, but with a messed up link order, where pseh is linked after ntdll, resulting in link-errors. The fix was to remove the link to pseh and add the include dir instead.
Additionally excpt.h will be included in other modules as well, requiring the include path. It just makes things easier.

Comment (learn-more) [reply CMakeLists.txt:328]: ah, that is very bad of gcc :(

**Review by learn-more — APPROVED** [review]
